### PR TITLE
ci: unified crates.io release workflow using release-plz

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
## Overview

Use a common unified CI for Rust release using [release-plz](https://release-plz.dev/).

> [Unified workflow source code](https://github.com/matter-labs/zksync-ci-common/blob/main/.github/workflows/release-plz.yaml)

## Details

* The workflow is similar to the `release-please` workflow as it creates a release PR with expected changes, monitors next semver, updates changelogs etc. but it's more specialized for Rust packages because it supports `crates.io` publishing out of the box, additional breaking API checks using `cargo-semver-checks`, easier to configure and support (for example, only one workflow file instead of two).
* The behavior is configurable through `release-plz.toml` file in the workspace root. Comments added to each line to explain what each setting does.

## Current restrictions

As `release-plz` does not provide the same output context as `release-please`, notifications for successful releases are not sent yet in the same format. It would require refactoring of the `format-release-please-for-slack-action` or simplier, just use GitHub release notifications in Slack that will not require any set up.

### CODE REVIEW

File: release-plz.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[workspace]
+dependencies_update = true                   # Wether to run `cargo update` in the release PR
+pr_name = "chore(main): release {{version}}" # template for the PR name
+pr_labels = ["release", "crates.io"]         # Set PR labels
+changelog_path = "./CHANGELOG.md"            # Set common changelog path for all crates
+changelog_update = false                     # Disable individual changelog updates for all packages.
+git_tag_enable = false                       # Disable individual tags for all packages.
+git_release_enable = false                   # Disable individual releases for all packages.
+semver_check = true                          # Enable API breaking changes checks with cargo-semver-checks.
```
Comment (popzxc) [inline release-plz.toml:9]: Should be false

**Review by popzxc — COMMENTED** [review]
I guess [RELEASE.md](https://github.com/matter-labs/zksync-protocol/blob/main/RELEASE.md) also needs to be updated.

File: release-plz.toml (hunk 1 above)
Comment (antonbaliasnikov) [reply release-plz.toml]: Fixed to `false` and updated RELEASE.md with `release-plz` link. I believe the other content is still valid.

**Comment by antonbaliasnikov** [comment]
@popzxc , I also checked that the bot user has the correct permissions for this repository, so we can proceed with the merge and the final stage of testing.

**Review by popzxc — APPROVED** [review]

**Comment by antonbaliasnikov** [comment]
Some more fixes are in progress.

**Review by popzxc — APPROVED** [review]

**Pull request merged**